lib/object/filter.js

```diff
@@ -8,8 +8,13 @@
 const typeOf = require('../type-of.js')
 module.exports = filter
 
-function filter(obj, fn) {
-  const out = Object.create(null)
+function filter(obj, fn, use_null_prototype = true) {
+  let out
+  if (use_null_prototype) {
```

Comment:
Had to use an old style since existential checks like `??` and `?` weren't around until node 14.  It seemed easy enough to not have to force `strictEngines` here and do a major.

Reply:
Gonna close this in favor of a `major`, but to answer the question, it's just because the argument would have to be an object literal created by the caller.  It seemed like a lot more error checking to do it in a way that wasn't a boolean.  Moot point, though.